object-curly-spacing possible bug with destructuring and trailing commas

I have the following code:

```js
const testing = { test: 1 };
let {
  test,
} = testing;
                               
test++;
```

With the new object-curly-spacing rule set to `always` in ESLint 0.22.1, I see the following output:

> 3:6  error  A space is required before ','  object-curly-spacing

I wasn't able to find any mention of this in the documentation for the rule, so I'm not sure if it was intentional or not, but it seems a bit strange to me so I thought it might be a bug.

Thanks for all of your great work on this tool!

Here's what I get with `0.22.1`:

```
  1:16  error  There should be no space after '{'   object-curly-spacing
  1:26  error  There should be no space before '}'  object-curly-spacing
```

--- a/lib/rules/object-curly-spacing.js
+++ b/lib/rules/object-curly-spacing.js
@@ -149,6 +149,12 @@ module.exports = function(context) {
                 penultimate = context.getLastToken(lastSpecifier),
                 last = context.getTokenAfter(lastSpecifier);
 
+            // support trailing commas
+            if (last.value === ",") {
+                penultimate = last;
+                last = context.getTokenAfter(last);
+            }
+
             validateBraceSpacing(node, first, second, penultimate, last);
         },
 

--- a/tests/lib/rules/object-curly-spacing.js
+++ b/tests/lib/rules/object-curly-spacing.js
@@ -35,6 +35,8 @@ eslintTester.addRuleTest("lib/rules/object-curly-spacing", {
         { code: "var {\nx,y\n} = z", options: ["always"], ecmaFeatures: { destructuring: true } },
         { code: "var { x = 10, y } = y", options: ["always"], ecmaFeatures: { destructuring: true } },
         { code: "var { x: { z }, y } = y", options: ["always"], ecmaFeatures: { destructuring: true } },
+        { code: "var {\ny,\n} = x", options: ["always"], ecmaFeatures: { destructuring: true } },
+        { code: "var { y, } = x", options: ["always"], ecmaFeatures: { destructuring: true } },
 
         // always - import / export
         { code: "import { door } from 'room'", options: ["always"], ecmaFeatures: { modules: true } },
@@ -70,6 +72,8 @@ eslintTester.addRuleTest("lib/rules/object-curly-spacing", {
         { code: "var {x = 10, y} = y", options: ["never"], ecmaFeatures: { destructuring: true } },
         { code: "var {x: {z}, y} = y", options: ["never"], ecmaFeatures: { destructuring: true } },
         { code: "var {\nx: {z\n}, y} = y", options: ["never"], ecmaFeatures: { destructuring: true } },
+        { code: "var {\ny,\n} = x", options: ["never"], ecmaFeatures: { destructuring: true } },
+        { code: "var {y,} = x", options: ["never"], ecmaFeatures: { destructuring: true } },
 
         // never - import / export
         { code: "import {door} from 'room'", options: ["never"], ecmaFeatures: { modules: true } },
@@ -154,6 +158,34 @@ eslintTester.addRuleTest("lib/rules/object-curly-spacing", {
                     type: "ObjectExpression",
                     line: 1,
                     column: 60
+                }
+            ]
+        },
+
+        // always-destructuring trailing comma
+        {
+            code: "var { a,} = x;",
+            options: ["always"],
+            ecmaFeatures: { destructuring: true },
+            errors: [
+                {
+                    message: "A space is required before '}'",
+                    type: "ObjectPattern",
+                    line: 1,
+                    column: 8
+                }
+            ]
+        },
+        {
+            code: "var {a, } = x;",
+            options: ["never"],
+            ecmaFeatures: { destructuring: true },
+            errors: [
+                {
+                    message: "There should be no space before '}'",
+                    type: "ObjectPattern",
+                    line: 1,
+                    column: 8
                 }
             ]
         },

